docs/README.md

```diff
@@ -1,2 +1,294 @@
+
+**Original documentation**
+Original documentation can be found [here](https://github.com/johnwhitton/horizon/tree/main/cli#ethereum-block-relay-cli) and says
+
+> Before using the CLI, fill in the private key into `.env` and execute `source .env`.
+1. `node index.js ethRelay getBlockHeader` get block header from ethereum.
+```
+node index.js ethRelay getBlockHeader https://ropsten.infura.io/v3/<project-id> 11266872
+```
+2. `node index.js ethRelay relay` constantly relay blocks from Ethereum to Harmony.
+
+**Questions**
+
+1. Can we simply run `node index.js ethRelay relay` to run the relayer?
+2. Are there any other components which need to be run?
+3. Are there any other infrastrcture tasks which need to be setup prior to running the relayer?
+
+
+## End to End Testing
+End to End testing currently consists of mapping ERC20 tokens from Harmony to Ethereum and transferring them accross the chains. This is done via the [ClI](https://github.com/johnwhitton/horizon/tree/main/cli) using the following commands
+
+5. `node index.js Bridge map` map ERC20 from ethereum to harmony.
+6. `node index.js Bridge crossTo` cross transfer ERC20 from ethereum to harmony.
+7. `node index.js Bridge crossBack` cross transfer HRC20 from harmony back to ethereum.
+
+There are also two scripts written to facilitate testing. 
+1. [test.js](https://github.com/johnwhitton/horizon/blob/main/scripts/test.js)
+2. [newtests.js](https://github.com/johnwhitton/horizon/blob/main/scripts/newtest.js)
+
+Also [end2end.js](https://github.com/johnwhitton/horizon/blob/main/scripts/end2end.js) which does not have any code in it.
+
+**Original Documentation**
+Original documentation for the CLI contains [Bridge CLI](https://github.com/johnwhitton/horizon/tree/main/cli#bridge-cli), [Ethereum Receipt Prove CLI](https://github.com/johnwhitton/horizon/tree/main/cli#ethereum-receipt-prove-cli) and [Ethereum Receipt Verifier CLI](https://github.com/johnwhitton/horizon/tree/main/cli#ethereum-receipt-verifier-cli).
+
+The assumption is that you need to just run the map, crossTo and crossBack functions.
+
+**Questions**
+1. Can testing be done via the CLI using he map, crossTo and crossBack functions?
+2. What additional setup or components need to be installed/run?
+3. Has end to end testing been completed?
+4. What outstanding development tasks are there? 
+
+
+## Smart Contract Testing
+Smart contract testing has a [test folder](https://github.com/johnwhitton/horizon/tree/main/test) which contains [bridg.hmy.js](https://github.com/johnwhitton/horizon/blob/main/test/bridge.hmy.js) however at the time of writing this test was not working.
+
+Tests are set up in the tests folder and are run using hardhat.
+
+`yarn test` is set up as a script to run `npx hardhat test`
+
+**Questions**
+1. Is there any other branches where tests have been written?
```

Comment:
Some PRs seem to be introducing useful tests, for example: https://github.com/harmony-one/horizon/pull/31/files